### PR TITLE
test: Add stickies tests

### PR DESCRIPTION
**Pull request by mutdmour**
Github issue / Community forum post (link here to close automatically):

### CODE REVIEW

File: cypress/e2e/17-stickies.cy.ts [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+	it('adds sticky to canvas with default text and position', () => {
+		workflowPage.getters.addStickyButton().should('not.be.visible');
+		workflowPage.actions.addSticky();
+
+		workflowPage.getters.stickies().should('have.length', 1)
+			.and(($el) => {
+				expect($el).to.have.css('top', '340px');
+				expect($el).to.have.css('left', '400px');
+				expect($el).to.have.css('height', '160px');
+				expect($el).to.have.css('width', '240px');
```
Comment (ivov) [inline cypress/e2e/17-stickies.cy.ts:22]: Default position and default dimensions are often repeated so we could generalize.

File: cypress/e2e/17-stickies.cy.ts [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+				expect($el).to.have.css('width', '346px');
+			});
+
+		cy.drag('[data-test-id="sticky"] [data-dir="right"]', [-50, -50]);
+		workflowPage.getters.stickies()
+			.should(($el) => {
+				expect($el).to.have.css('top', '340px');
+				expect($el).to.have.css('left', '400px');
+				expect($el).to.have.css('height', '160px');
+				expect($el).to.have.css('width', '302px');
```
Comment (ivov) [inline cypress/e2e/17-stickies.cy.ts:109]: The difference is not exactly drag length because of margin/padding?

**Review by ivov — DISMISSED** [review]
Nice! All passing locally. Maybe a followup would be to ensure a sticky cannot cover a node when they overlap.

File: cypress/e2e/17-stickies.cy.ts (hunk 1 above)
Comment (mutdmour) [reply cypress/e2e/17-stickies.cy.ts:22]: fair.. will try to see if I could standarize this..

File: cypress/e2e/22-stickies.cy.ts [hunk 3]
```diff
@@ -0,0 +1,343 @@
+import { WorkflowPage as WorkflowPageClass } from '../pages/workflow';
+
+const workflowPage = new WorkflowPageClass();
+
+describe('Canvas Actions', () => {
+	beforeEach(() => {
+		cy.resetAll();
+		cy.skipSetup();
+		workflowPage.actions.visit();
+		cy.waitForLoad();
```
Comment (OlegIvaniv) [inline cypress/e2e/22-stickies.cy.ts:10]: There's no need to `waitForLoad` here, it's already part of the `workflowPage.actions.visit`

File: cypress/e2e/22-stickies.cy.ts [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+				expect($el).to.have.css('left', '466px');
+				expect($el).to.have.css('height', '160px');
+				expect($el).to.have.css('width', '194px');
+			});
+	});
+
+	it('expands/shrinks sticky from the top edge', () => {
+		workflowPage.actions.addSticky();
+		cy.drag('[data-test-id="sticky"]', [100, 100]); // move away from canvas button
+		workflowPage.getters.stickies()
```
Comment (OlegIvaniv) [inline cypress/e2e/22-stickies.cy.ts:100]: We could save **a lot** of lines here if we'd wrap this check as a function:
```
function checkStickiesStyle(top: number, left: number, height: number, width: number, zIndex?: number) {
	workflowPage.getters.stickies().should(($el) => {
		expect($el).to.have.css('top', `${top}px`);
		expect($el).to.have.css('left', `${left}px`);
		expect($el).to.have.css('height', `${height}px`);
		expect($el).to.have.css('width', `${width}px`);
		if (zIndex) {
			expect($el).to.have.css('z-index', `${zIndex}`);
		}
	});
}
```

File: cypress/e2e/22-stickies.cy.ts [hunk 5]
```diff
@@ -0,0 +1,343 @@
...
+describe('Canvas Actions', () => {
+	beforeEach(() => {
+		cy.resetAll();
+		cy.skipSetup();
+		workflowPage.actions.visit();
+		cy.waitForLoad();
+
+		cy.window()
+		// @ts-ignore
+		.then(win => win.onBeforeUnload && win.removeEventListener('beforeunload', win.onBeforeUnload))
```
Comment (OlegIvaniv) [inline cypress/e2e/22-stickies.cy.ts:14]: We changed this recently to `onBeforeUnloadNodeView`. Maybe we could extract this to cy.command so there's a single source of truth

**Review by OlegIvaniv — CHANGES_REQUESTED** [review]

File: cypress/e2e/22-stickies.cy.ts (hunk 5 above)
Comment (mutdmour) [reply cypress/e2e/22-stickies.cy.ts]: updated now to the latest approach.

**Review by OlegIvaniv — APPROVED** [review]
Looks good to me!

**Comment by janober** [comment]
Got released with [`n8n@0.226.1`](https://github.com/n8n-io/n8n/releases/tag/n8n%400.226.1)

**Pull request merged**